assets/src/components/Snapping.js

```diff
@@ -37,6 +37,20 @@ export default class Snapping extends HTMLElement {
                     </div>
                 </div>
             </div>
+            ${mainLizmap.snapping.active ?
+                html`<div class="control-group">
```

Comment:
Please do not use the same level for the existing "Snapping" ( `<h3>` ) and the new "Snap on". For me, the latter belongs to the div Snapping. It is a new configuration option and should be put just under the buttons Stop & refresh, with no use of `h3` , and possibly in the same `<div>` as the existing Snapping group.

Please also add a `title` attribute to the new items to let the user read a tooltip by hovering the mouse

* Possible generic title when hovering the layer title (and the checkbox before): `Toggle snapping for this layer`
* For disabled layers, a precise enough sentence explaining why the layer is disabled and how to enable snapping for it